### PR TITLE
Release: 2.19

### PR DESCRIPTION
**Pull request by neSpecc**
- `New` - Read-only mode 🥳 [#837](https://github.com/codex-team/editor.js/issues/837)
- `New` - RTL mode added [#670](https://github.com/codex-team/editor.js/issues/670)
- `Fix` — Fix problem with types usage [#1183](https://github.com/codex-team/editor.js/issues/1183)
- `Fix` - Fixed issue with Spam clicking the "Click to tune" button duplicates the icons on FireFox. [#1273](https://github.com/codex-team/editor.js/issues/1273)
- `Fix` - Fixed issue with `editor.blocks.delete(index)` method which throws an error when Editor.js is not focused, even after providing a valid index. [#1182](https://github.com/codex-team/editor.js/issues/1182)
- `Improvements` - The `initialBlock` property of Editor config is deprecated. Use the `defaultBlock` instead. [#993](https://github.com/codex-team/editor.js/issues/993)
- `Fix` - Fixed the issue of toolbar not disappearing on entering input in Chinese, Hindi and some other languages. [#1196](https://github.com/codex-team/editor.js/issues/1196)
- `Improvements` - BlockAPI `call()` method now returns the result of calling method, thus allowing it to expose arbitrary data as needed [#1205](https://github.com/codex-team/editor.js/pull/1205)
- `Improvements` - Unuseful log about missed i18n section has been removed [#1269](https://github.com/codex-team/editor.js/issues/1269)
- `Fix` - Do not stop events propagation if not needed (essential for React synthetic events) [#1051](https://github.com/codex-team/editor.js/issues/1051) [#946](https://github.com/codex-team/editor.js/issues/946)
- `Fix` - Tool's `destroy` method is not invoked when `editor.destroy()` is called. [#1047](https://github.com/codex-team/editor.js/issues/1047)
- `New` - Tool's `reset` static method added to the API to clean up any data added by Tool on initialization
- `Improvements` - Unuseful log about missed i18n section has been removed  [#1269](https://github.com/codex-team/editor.js/issues/1269)
- `Fix` - Fixed issue with enter key in inputs and textareas [#920](https://github.com/codex-team/editor.js/issues/920)
- `Improvements` - Allowed to set `false` as `toolbox` config in order to hide Toolbox button [#1221](https://github.com/codex-team/editor.js/issues/1221)
- `New` - Allows users to provide common `inlineToolbar` property which will be used for all tools whose `inlineToolbar` property is set to `true`. It can be overridden by the tool's own `inlineToolbar` property. Also, inline tools will be ordered according to the order of the inline tools in array provided in the `inlineToolbar` property. [#1056](https://github.com/codex-team/editor.js/issues/1056)
- `Fix` - blocks.getBlockByIndex() API method now returns void for indexes out of range [#1270](https://github.com/codex-team/editor.js/issues/1270)
- `Fix` - Fixed the `onChange` callback issue. This method didn't be called for native inputs before some contentedtable element changed [#843](https://github.com/codex-team/editor.js/issues/843)

### CODE REVIEW

File: src/components/block-tunes/block-tune-move-up.ts [hunk 1]
```diff
@@ -82,8 +82,13 @@ export default class MoveUpTune implements BlockTune {
     }
 
     const currentBlock = this.api.blocks.getBlockByIndex(currentBlockIndex);
-    const currentBlockElement = currentBlock.holder;
     const previousBlock = this.api.blocks.getBlockByIndex(currentBlockIndex - 1);
+
+    if (!currentBlock || !previousBlock) {
```
Comment (neSpecc) [inline src/components/block-tunes/block-tune-move-up.ts:87]: can we merge this statement with the previous one?

File: types/tools/block-tool.d.ts [hunk 2]
```diff
@@ -16,6 +16,11 @@ export interface BlockTool extends BaseTool {
    */
   sanitize?: SanitizerConfig;
 
+  /**
+   * Creates and returns Tools HTML markup
+   */
+  render(): HTMLElement;
```
Comment (neSpecc) [inline types/tools/block-tool.d.ts:22]: this is not needed here.

**Review by gohabereg — APPROVED** [review]

**Comment by neSpecc** [comment]
Block Shortcuts are triggering twice.

**Comment by neSpecc** [comment]
Block Shortcuts can paste blocks in Read-only

**Review by khaydarov — APPROVED** [review]

**Pull request merged**